python/mpt/tests/src/mpt/test_utils.py

```diff
@@ -74,3 +77,66 @@ def test_check_branch_node_invalid_x_branch(
         ):
             with strict_raises(ValueError):
                 cairo_run_py("check_branch_node", branch_node)
+
+    @given(data=...)
+    def test_sort_account_diff(
+        self, cairo_run, data: List[AddressAccountNodeDiffEntry]
+    ):
+        sorted_data = sorted(
+            data, key=lambda x: int.from_bytes(x.key, "little"), reverse=True
+        )
+        cairo_data = cairo_run("sort_account_diff", data)
+        assert cairo_data == sorted_data
+
+    @given(data=...)
+    def test_sort_account_diff_ascending_order(
+        self, cairo_programs, cairo_run_py, data: List[AddressAccountNodeDiffEntry]
+    ):
+        with patch_hint(
+            cairo_programs,
+            "sort_account_diff",
+            """
+# Extract the list of pointers directly
+pointers = [memory[ids.diffs_ptr.address_ + i] for i in range(ids.diffs_len)]
+
+# Sort pointers based on the key values they point to, in ascending order
+sorted_pointers = sorted(pointers, key=lambda ptr: memory[ptr])
+
+# Load the sorted pointers into ids.buffer
+segments.load_data(ids.buffer, sorted_pointers)
+
+indices = list(range(ids.diffs_len))
+sorted_indices = sorted(indices, key=lambda i: memory[pointers[i]], reverse=True)
+segments.load_data(ids.sorted_indexes, sorted_indices)
+            """,
+        ):
+            with strict_raises(ValueError):
+                cairo_run_py("sort_account_diff", data)
+
+    @given(data=...)
+    def test_sort_account_diff_different_lists(
+        self, cairo_programs, cairo_run_py, data: List[AddressAccountNodeDiffEntry]
+    ):
+        with patch_hint(
+            cairo_programs,
+            "sort_account_diff",
+            """
+# Extract the list of pointers directly
+pointers = [memory[ids.diffs_ptr.address_ + i] for i in range(ids.diffs_len)]
+
+# Sort pointers based on the key values they point to, in ascending order
+sorted_pointers = sorted(pointers, key=lambda ptr: memory[ptr])
+
+# Load the sorted pointers into ids.buffer
+# Repeat the first element n times to create a different list
+first_element = sorted_pointers[0]
+repeated_list = [first_element] * ids.diffs_len
+segments.load_data(ids.buffer, repeated_list)
+
+indices = list(range(ids.diffs_len))
+sorted_indices = sorted(indices, key=lambda i: memory[pointers[i]], reverse=True)
+segments.load_data(ids.sorted_indexes, sorted_indices)
```

Comment:
```suggestion
    pointers = [memory[ids.diffs_ptr.address_ + i] for i in range(ids.diffs_len)]
    sorted_pointers = sorted(pointers, key=lambda ptr: memory[ptr], reverse=True)
    # BAD HINT: not a permutation of the input list
    sorted_pointers[-1] += 1
    segments.load_data(ids.buffer, sorted_pointers)
    indices = list(range(ids.diffs_len))
    sorted_indices = sorted(indices, key=lambda i: memory[pointers[i]], reverse=True)
    segments.load_data(ids.sorted_to_original_index_map, sorted_indices)
```

Make the change targeted and explicit